packages/page-migration/src/Migration/index.tsx

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017-2022 @polkadot/app-rpc authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect } from 'react';
+
+import type { DeriveBalancesAll } from '@polkadot/api-derive/types';
+
+import { Available, Dropdown, InputAddressCheqd } from '@polkadot/react-components';
+import { InputAddress, MarkError, MarkWarning, TxButton, Button } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { isValidCheqdAddress, getCheqdAddressError } from '@polkadot/react-components/InputAddressCheqd';
+import { useApi, useCall } from '@polkadot/react-hooks';
+import { formatBalance } from '@polkadot/util';
+
+function convertDockToCheqd(dockBalance) {
+  if (!dockBalance) {
+    return 0;
+  }
+  const swapRatio = 18.5178;
+  const dockAmt = dockBalance.toNumber() / 1000000;
+  const cheqdBalance = dockAmt / swapRatio;
+  return cheqdBalance;
+}
+
+function hasAvailableBalance(dockBalance) {
+  if (!dockBalance || !dockBalance.availableBalance) {
+    return false;
+  }
+  return !dockBalance.availableBalance.isZero();
+}
+
+function hasLockedBalance(dockBalance) {
+  if (!dockBalance || !dockBalance.lockedBalance) {
+    return false;
+  }
+  return !dockBalance.lockedBalance.isZero();
+}
+
+function MigrationApp ({ className }): React.ReactElement {
+  const { t } = useTranslation();
+  const [senderId, setSenderId] = useState<string | null>(null);
+  const [cheqdId, setCheqdId] = useState<string>('');
+  const [selectFromDropdown, setSelectFromDropdown] = useState<boolean>(false);
+  const [cheqdAddresses, setCheqdAddresses] = useState<string[]>([]);
+  const { api } = useApi();
+  const allBalances = useCall<DeriveBalancesAll>(api.derive.balances?.all, [senderId]);
+
+  const values = [cheqdId];
+  const extrinsic = api && api.tx && api.tx.cheqdMigration && api.tx.cheqdMigration.migrate(values);
```

Comment:
I'm unsure if this is the best way to do this as it will create a transaction on every render.
It is probably worth caching it using `useMemo` based on provided `values`.

Reply:
in this case changing it every render makes sense, because the only times render is triggered is: page load, change dock account, change cheqd account